### ISSUE TITLE
horse or human inceptionv3 miscatagorizing

### ISSUE THREAD

**Issue by tameyer1** [lmoroney/tfbook#8]
Hello

I was working on the horse or human transfer learning section. I am using the datasets for training and validation from the urls you supply. No matter what horse image I try the model classifies it as human.  I figured my local setup might have issues so I opened and ran the colab file for transfer learning in the chapter 3 folder.  I received the same incorrect results using a random horse picture from the web as well as when I uploaded some of the validation horse images from the colab project.

**Comment by Rutraz** [comment]
Hey

I was having a similar issue. I realized that the images of horses and humans used to train the model on the first iteration of chapter 3 were 300 x 300 and later in the chapter when we arrive at transfer learning the inputs were 150 x 150. So I altered the code given and put it back to 300 x 300 and that worked for me.

![Inkedchrome_susClQU6im_LI](https://user-images.githubusercontent.com/44867933/112347759-0d04e980-8cbf-11eb-85f2-a37fd9a4ccbc.jpg)

**Comment by tameyer1** [comment]
I had been trying it the opposite way of setting the training, validation, and test images to 150,150.  I went ahead and tested with 300,300 in both the model training and test and receive the same result: running the validation images against the model always returns human and 1. for the class.  So still no luck but thanks for the reply.

**Comment by emin** [comment]
yes, I've the same problem as well. Using 300 adjustment didn't work either. I don't know the issue, considering I've started with this book to my ML journey, I'd like to know what's wrong with this.

@lmoroney

**Comment by bartpotrykus** [comment]
@tameyer1 Hey, I know your question is one year old but I encountered same issue. The reason is we were not rescaling when using a test image.
`x = x / 255.0`
is the line that made the classification correct.

**Comment by semajson** [comment]
I hit the exact same issue.

It looks like this might be a bug in: https://github.com/lmoroney/tfbook/blob/master/chapter3/multiclass.ipynb

where rather than:
```
for fn in uploaded.keys():
 
  # predicting images
  path = fn
  img = image.load_img(path, target_size=(150, 150))
  x = image.img_to_array(img)
  x = np.expand_dims(x, axis=0)

  images = np.vstack([x])
  classes = model.predict(images, batch_size=10)
  print(fn)
  print(classes)
```

it should be:
```
for fn in uploaded.keys():
 
  # predicting images
  path = fn
  img = image.load_img(path, target_size=(150, 150))
  x = image.img_to_array(img)
  x = np.expand_dims(x, axis=0)
  x = x / 255.0

  images = np.vstack([x])
  classes = model.predict(images, batch_size=10)
  print(fn)
  print(classes)
```

However, difference that might explain why i needed to add `x = x / 255.0` on my machine
-  I'm running on my WSL env, using `tensorflow-cpu` (version 2.11.0)
- I had to replace `img = image.load_img(path, target_size=(150, 150))` with `img = tf.keras.utils.load_img(test_image, target_size=(300, 300))` to get my code to not error
- i had to replace `x = image.img_to_array(img)`  with `x = tf.keras.utils.img_to_array(img)` to get my code to not error

I think the next step to figure out if this is an error in https://github.com/lmoroney/tfbook/blob/master/chapter3/multiclass.ipynb is to run that code on colab code to see if `x = x / 255.0` is needed

**Comment by semajson** [comment]
I ran the https://github.com/lmoroney/tfbook/blob/master/chapter3/transfer_learning.ipynb on colab, and needed to make these following changes:
Changes:
	- import tensorflow as tf
	- img = tf.keras.utils.load_img(path, target_size=(150, 150))
	- x = tf.keras.utils.img_to_array(img)
        - x = x / 255.0

I will raise an MR to fix

**Comment by semajson** [comment]
@lmoroney looks like people can't just raise PRs against this repo.

What is the correct to get this issue fixed?

**Comment by semajson** [comment]
Ah i just realised what i was doing wrong - leave this with me and i will raise a PR
